guides/10_Video_Processor.md

```diff
@@ -0,0 +1,115 @@
+
+After the call, the inner `Device` will be released by device controller unless the inner `Device` is a `MediaStream` provided by users and it is the responsibility of users to handle the lifecycle. 
+
+After `DefaultVideoTransformDevice` is no longer used by device controller, call `DefaultVideoTransformDevice.stop` to release the `VideoProcessor`s and underlying pipeline. After `stop` is called, users must discard the `DefaultVideoTransformDevice`.`DefaultVideoTransformDevice.stop` is necessary to release all the resources.
+
+
+#### Receiving lifecycle notifications with observer
+
+To receive notifications of lifecycle events, `DefaultVideoTransformDeviceObserver` can be added via `addObserver` method. 
+The full list of the callbacks:
+
+1. `processingDidStart`
+
+`processingDidStart` will be called when video processing starts.
+
+2. `processingDidFailToStart`
+
+`processingDidFailToStart` will be called when video processing could not start due to runtime errors. In this case, developers are expected to call `chooseVideoInputDevice` again with a valid `VideoInputDevice` to continue video sending. 
+
+3. `processingDidStop`
+
+`processingDidStop` will be called when video processing is stopped **expectedly**.
+
+4. `processingLatencyTooHigh(latencyMs: number)` 
+
+`processingLatencyTooHigh` will be called when the execution of processors slows the frame rate down by at least half.
+
+
+### VideoFrameProcessorPipeline
+The interface `VideoFrameProcessorPipeline` represents a video processing pipeline. The management of `VideoFrameProcessorPipeline` is already handled under the hood when `DefaultVideoTransformDevice` is used.
+
+
+#### Starting and Stopping
+To start the processing, call `DefaultVideoFrameProcessorPipeline.setInputMediaStream` with an active `MediaStream` and call `DefaultVideoFrameProcessorPipeline.getActiveOutputMediaStream` to obtain the post-processed `MediaStream`.
+
+The input `MediaStream` is constructed as a `VideoFrameBuffer` to be fed into the first `VideoFrameProcessor`.
+The last processor in the chain must return one `VideoFrameBuffer` so the pipeline could interpret the output as [CanvasImageSource](https://developer.mozilla.org/en-US/docs/Web/API/CanvasImageSource).
+
+The pipeline invokes each processor at an interval specified by the setter `framerate`. If the `framerate` is set to be higher than the intrinsic frame rate of video, processor might be wasting cycles re-drawing the same picture.
+
+
+To stop the processing, call `DefaultVideoFrameProcessorPipeline.stop`. To fully destroy the pipeline resources, call `DefaultVideoFrameProcessorPipeline.destroy`. 
+
+
+### VideoFrameBuffer and VideoFrameProcessor
```

Comment:
Could we add a quick sample here on how to use this (like how we did in the meeting demo app).

Reply:
added.